pydantic_ai_slim/pydantic_ai/_system_prompt.py

```diff
@@ -20,15 +20,15 @@ def __post_init__(self):
         self._takes_ctx = len(inspect.signature(self.function).parameters) > 0
         self._is_async = inspect.iscoroutinefunction(self.function)
 
-    async def run(self, run_context: RunContext[AgentDepsT]) -> str:
+    async def run(self, run_context: RunContext[AgentDepsT]) -> Union[str, None]:  # noqa UP007
         if self._takes_ctx:
             args = (run_context,)
         else:
             args = ()
 
         if self._is_async:
-            function = cast(Callable[[Any], Awaitable[str]], self.function)
+            function = cast(Callable[[Any], Awaitable[Union[str, None]]], self.function)
```

Comment:
Yeah, this is a place where you need `Union[str, None]`, though I think it would probably work to just use the string (i.e., `'str | None'`)